Content.Client/UserInterface/Controls/RadialMenu.cs

```diff
@@ -171,45 +179,37 @@ public void ReturnToPreviousLayer()
             ContextualButton.SetOnlyStyleClass(CloseButtonStyleClass);
     }
 }
-
-[Virtual]
-public class RadialMenuButton : Button
+public sealed class RadialMenuContextualCentralTextureButton : TextureButton
 {
-    /// <summary>
-    /// Upon clicking this button the radial menu will transition to the named layer
-    /// </summary>
-    public string? TargetLayer { get; set; }
-
-    /// <summary>
-    /// A simple button that can move the user to a different layer within a radial menu
-    /// </summary>
-    public RadialMenuButton()
-    {
-        OnButtonUp += OnClicked;
-    }
 
-    private void OnClicked(ButtonEventArgs args)
+    /// <inheritdoc />
+    public RadialMenuContextualCentralTextureButton()
     {
-        if (TargetLayer == null || TargetLayer == string.Empty)
-            return;
 
-        var parent = FindParentMultiLayerContainer(this);
+    }
 
-        if (parent == null)
-            return;
+    public RadialContainer? Container { get; set; }
 
-        parent.TryToMoveToNewLayer(TargetLayer);
+    /// <inheritdoc />
+    public override bool IsPositionInside(Vector2i point)
+    {
+        return HasPoint(point);
     }
 
-    private RadialMenu? FindParentMultiLayerContainer(Control control)
+    /// <inheritdoc />
+    protected override bool HasPoint(Vector2 point)
     {
-        foreach (var ancestor in control.GetSelfAndLogicalAncestors())
+        if (Container == null)
```

Comment:
to properly draw it considering radial menu button sectors - we need at least container itself (which contains radius data and can be different for each 'level' of menu).